### PR TITLE
[docs] [tune] replace tune.report with session.report

### PR DESCRIPTION
**Pull request by angelinalg**
## Why are these changes needed?

To address CSAT reported feedback that example was using an outdated API (tune.report) which is replaced with session.report.

## Related issue number

(https://github.com/ray-project/ray/issues/34434)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/tests/tutorial.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def train_mnist(config):
         acc = test(model, test_loader)
 
         # Send the current training result back to Tune
-        tune.report(mean_accuracy=acc)
+        session.report(mean_accuracy=acc)
```
Comment (matthewdeng) [inline python/ray/tune/tests/tutorial.py:108]: Will need to also include the proper import at the top: 

`from ray.air import session`

**Review by matthewdeng — APPROVED** [review]
Code looks good to me. Test is failing for an unrelated reason though...

**Comment by angelinalg** [comment]
@matthewdeng I've never merged a PR for this directory before. Are you able to merge it? Or should I rebase and try to get the tests to pass, first?

**Comment by matthewdeng** [comment]
@richardliaw can you merge? If somehow this breaks master I will take full responsibility.

**Comment by krfricke** [comment]
> @richardliaw can you merge? If somehow this breaks master I will take full responsibility.

famous last words :-D

Screenshot 2023-04-21 at 9 53 56 AM

I'll file a fix

**Comment by matthewdeng** [comment]
Lol sigh, thanks @krfricke 😅

**Pull request merged**